message/retrylater.go

```diff
@@ -21,10 +22,19 @@ type retryLaterHandler struct {
 }
 
 func (r *retryLaterHandler) Do(ctx context.Context, orig Handler, message *servicebus.Message) Handler {
-	select {
-	case <-ctx.Done():
-		return Error(fmt.Errorf("aborting retrylater for message. abandoning meddage %s: %w", message.ID, ctx.Err()))
-	case <-time.After(r.retryAfter):
-		return orig
-	}
+	go func() {
+		ctx, span := startSpanFromMessageAndContext(ctx, "go-shuttle.retryLater.Do", message)
+		defer span.End()
+
+		select {
+		//TODO this can go past lock duration pretty easily
```

Comment:
Feels concerning to leave this, but I think its ok to complete. Its much better than before :)

Reply:
Agree. Lets create an issue on the github too. I think for RP we want to go back to a model where we constantly renew in the background.